### ISSUE TITLE
antsApplyTransforms allows `-d 4` with a 3D warp field

### ISSUE THREAD

**Issue by cookpa** [ANTsX/ANTs#1721]
As reported in #1717

```
antsApplyTransforms -d 4 -r fixed.nii.gz -i moving.nii.gz -t antsregsyn_1Warp.nii.gz -o output.nii.gz --verbose
```

This will run without errors though the output will be wrong.

**Comment by cookpa** [comment]
Actually I might have fixed this already in #1694, will check

**Comment by cookpa** [comment]
OK confirmed, it still allows

```
antsApplyTransforms -d 4 -r fixed.nii.gz -i timeseries.nii.gz -t antsregsyn_1Warp.nii.gz -o output.nii.gz --verbose
```

where timeseries.nii.gz is 4D, but the warp is only 3D. But it only then outputs the first volume, so it will be obviously wrong.
